frontend/src/app/auth.service.ts

```diff
@@ -47,6 +47,28 @@ export class AuthService {
         }
       });
   }
+  resetPassword(username: {username: string}): Observable<string> {
+    return this.http.post(
+      this._passwordResetUrl, username)
+      .map(response => {
+        if (response.status === 201) {
+          return 'success';
+        }
+      }).catch(err => Observable.of('error'));
+  }
+
+  confirmResetPassword(
+    password: {password: string},
+    uidb64: string,
+    token: string): Observable<string> {
+    return this.http.post(
+      `${this._passwordResetUrl}/${uidb64}/${token}`, password)
```

Comment:
If password would be a string parameter, then on this line you wrap it in object {password} and send it to server.